### PR TITLE
samples: bluetooth: Interrupt driven DTM sample

### PR DESCRIPTION
**Pull request by KAGA164**
This PR provides the interrupt driven DTM sample

### CODE REVIEW

File: samples/bluetooth/direct_test_mode/src/dtm.c [hunk 1]
```diff
@@ -1924,3 +1771,179 @@ bool dtm_event_get(uint16_t *dtm_event)
...
+
+static struct dtm_pdu *radio_buffer_swap(void)
+{
+	struct dtm_pdu *received_pdu = dtm_inst.current_pdu;
+
+	if (dtm_inst.current_pdu == dtm_inst.pdu) {
+		dtm_inst.current_pdu = dtm_inst.pdu + 1;
+	} else {
+		dtm_inst.current_pdu = dtm_inst.pdu;
+	}
```
Comment (kapi-no) [inline samples/bluetooth/direct_test_mode/src/dtm.c:1783]: See suggestion from my previous review

File: samples/bluetooth/direct_test_mode/src/main.c [hunk 2]
```diff
@@ -82,15 +81,14 @@ void main(void)
 		/* This is the second byte read; combine it with the first and
 		 * process command.
 		 */
-		if (current_time >
-			(msb_time + MAX_ITERATIONS_NEEDED_FOR_NEXT_BYTE)) {
```
Comment (kapi-no) [inline samples/bluetooth/direct_test_mode/src/main.c:86]: Please remove this define and its description if its no longer needed

Comment (KAGA164) [reply samples/bluetooth/direct_test_mode/src/main.c:86]: Yes right. Removed

File: samples/bluetooth/direct_test_mode/src/dtm.c (hunk 1 above)
Comment (KAGA164) [reply samples/bluetooth/direct_test_mode/src/dtm.c]: Done

**Comment by joerchan** [comment]
This needs to be rebased now.

**Review by grochu — APPROVED** [review]
Looks good.

**Review by kapi-no — APPROVED** [review]

File: samples/bluetooth/direct_test_mode/src/dtm.c [hunk 3]
```diff
@@ -1808,10 +1637,13 @@ static enum dtm_err_code on_test_end_cmd(void)
...
 static enum dtm_err_code on_test_receive_cmd(void)
 {
+	dtm_inst.current_pdu = dtm_inst.pdu;
+
 	/* Zero fill all pdu fields to avoid stray data from earlier
 	 * test run.
 	 */
-	memset(&dtm_inst.pdu, 0, DTM_PDU_MAX_MEMORY_SIZE);
+	memset(&dtm_inst.pdu[0], 0, DTM_PDU_MAX_MEMORY_SIZE);
+	memset(&dtm_inst.pdu[1], 0, DTM_PDU_MAX_MEMORY_SIZE);
```
Comment (doki-nordic) [inline samples/bluetooth/direct_test_mode/src/dtm.c:1646]: Maybe safer and cleaner will be:
```
memset(&dtm_inst.pdu, 0, sizeof(dtm_inst.pdu));
```

**Review by doki-nordic — APPROVED** [review]

File: samples/bluetooth/direct_test_mode/src/dtm.c (hunk 3 above)
Comment (KAGA164) [reply samples/bluetooth/direct_test_mode/src/dtm.c]: Good idea. Done

**Pull request merged**